.semaphore/semaphore.yml

```diff
@@ -0,0 +1,176 @@
+version: v1.0
+name: CI Pipeline
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu2004
+global_job_config:
+  epilogue:
+    always:
+      commands:
+        - '[[ -f report.xml ]] && test-results publish report.xml'
+blocks:
+  - name: Setup
+    task:
+      env_vars:
+        - name: GO_VERSION
+          value: 1.16.7
+      jobs:
+        - name: Variables
+          commands:
+            - NPROC=$(nproc)
+            - GO_VERSION=1.19.4
+            - PROTOC_VERSION=21.12
+            - PROTOC_GEN_VERSION=v1.28.1
+            - PROTOC_GRPC_VERSION=v1.2.0
+        - name: Update Go
+          commands:
+            - |
+              function version_gt() {
+                test "$(printf ''%s\n'' "$@" | sort -V | head -n 1)" != "$1"
+              }
+            - |
+              CURRENT_GO_VERSION=$(go version | sed 's/[^0-9.]*\([0-9.]*\).*/\1/')
+            - |
+              if version_gt $GO_VERSION $CURRENT_GO_VERSION; then
+                echo "Updating go version from $CURRENT_GO_VERSION to $GO_VERSION ..."
+                sudo rm -rf /usr/bin/go
+                sudo rm -rf /usr/local/go
+                sudo rm -rf /usr/local/bin/go
+                sudo rm -rf /usr/local/golang
+                sudo rm -rf $GOROOT $GOPATH $GOBIN
+                wget https://go.dev/dl/go$GO_VERSION.linux-amd64.tar.gz
+                sudo tar -C /usr/local -xzf go$GO_VERSION.linux-amd64.tar.gz
+                export GOROOT=/usr/local/go
+                export PATH=$PATH:/usr/local/go/bin
+              fi
+            - export GOBIN=$HOME/go/bin
+            - 'export PATH=$PATH:$GOBIN'

```

Comment:
We probably do not need to update Go anymore. This was only used as a way of enabling the latest version of Go, but backward compatibility guarantees that it should be just fine if the CI Go version is 1 or 2 (but not more) older than the version in `go.mod`.